src/main/java/baritone/utils/schematic/format/defaults/LitematicaSchematic.java

```diff
@@ -0,0 +1,217 @@
+
+    public LitematicaSchematic(NBTTagCompound nbt) {
+        String regionName = (String) nbt.getCompoundTag("Regions").getKeySet().toArray()[0];
+        this.x = Math.abs(nbt.getCompoundTag("Regions").getCompoundTag(regionName).getCompoundTag("Size").getInteger("x"));
+        this.y = Math.abs(nbt.getCompoundTag("Regions").getCompoundTag(regionName).getCompoundTag("Size").getInteger("y"));
+        this.z = Math.abs(nbt.getCompoundTag("Regions").getCompoundTag(regionName).getCompoundTag("Size").getInteger("z"));
+        this.states = new IBlockState[this.x][this.z][this.y];
+
+
+        NBTTagList paletteTag = nbt.getCompoundTag("Regions").getCompoundTag(regionName).getTagList("BlockStatePalette",10);
+        // ListNBT paletteTag = nbt.getCompound("Regions").getCompound(regionName).getList("BlockStatePalette",10);
+
+        // Create the block states array
+        IBlockState[] paletteBlockStates = new IBlockState[paletteTag.tagCount()];
+        // For every part of the array
+        for (int i = 0; i<paletteTag.tagCount(); i++) {
+            // Set the default state by getting block name
+            Block block = Block.REGISTRY.getObject(new ResourceLocation((((NBTTagCompound) paletteTag.get(i)).getString("Name"))));
+            IBlockState blockState = block.getDefaultState();
+            NBTTagCompound properties = ((NBTTagCompound) paletteTag.get(i)).getCompoundTag("Properties");
+            Object[] keys = properties.getKeySet().toArray();
+            Map<String, String> propertiesMap = new HashMap<>();
+            // Create a map for each state
+            for (int j = 0; j<keys.length; j++) {
+                propertiesMap.put((String) keys[j], (properties.getString((String) keys[j])));
+            }
+            for (int j = 0; j<keys.length; j++) {
+                IProperty<?> property = block.getBlockState().getProperty(keys[j].toString());
+                if (property != null) {
+                    blockState = setPropertyValue(blockState, property, propertiesMap.get(keys[j]));
+                }
+            }
+            paletteBlockStates[i] = blockState;
+        }
+
+
+        // BlockData is stored as an NBT long[]
+        int paletteSize = (int) Math.floor(log2(paletteTag.tagCount()))+1;
+        long litematicSize = (long) this.x*this.y*this.z;
+
+        // In 1.12, the long array isn't exposed by the libraries so parsing has to be done manually
+        String rawBlockString = (nbt.getCompoundTag("Regions").getCompoundTag(regionName)).getTag("BlockStates").toString();
+        rawBlockString = rawBlockString.substring(3,rawBlockString.length()-1);
+        String[] rawBlockArrayString = rawBlockString.split(",");
+        long[] rawBlockData = new long[rawBlockArrayString.length];
+        for (int i = 0; i < rawBlockArrayString.length; i++) {
+            rawBlockData[i] = Long.parseLong(rawBlockArrayString[i].substring(0,rawBlockArrayString[i].length()-1));
+        }
```

Comment:
Since I hate parsing strings that aren't meant to be parsed, may I suggest using reflection or mixin to access the `data` attribute? I checked the litematica and malilib sources and the latter is what is used there.

Reply:
Yes so that's the main reason why I have this as a draft pull because that iteration is so tacky. I'll look into those sources and try to find something.